common/workflows/restricted.py

```diff
@@ -0,0 +1,239 @@
+from datetime import timedelta
+
+from invenio_rdm_records.services.generators import IfRestricted, IfDraft
+from invenio_records_permissions.generators import AnyUser, Disable
+from oarepo_communities.services.permissions.generators import (
+    CommunityRole,
+    PrimaryCommunityMembers,
+    PrimaryCommunityRole,
+)
+from oarepo_communities.services.permissions.policy import (
+    CommunityDefaultWorkflowPermissions,
+)
+from oarepo_requests.services.permissions.generators import IfRequestedBy
+from oarepo_runtime.services.permissions.generators import RecordOwners
+from oarepo_workflows import (
+    AutoApprove,
+    IfInState,
+    WorkflowRequest,
+    WorkflowRequestEscalation,
+    WorkflowRequestPolicy,
+    WorkflowTransitions,
+)
+from invenio_i18n import lazy_gettext as _
+
+
+class RestrictedWorkflowPermissions(CommunityDefaultWorkflowPermissions):
+    can_create = [
+        PrimaryCommunityRole("submitter"),
+        PrimaryCommunityRole("owner"),
+        PrimaryCommunityRole("curator"),
+    ]
+
+    can_read_generic = [
+        RecordOwners(),
+        CommunityRole("curator"),
+        CommunityRole("owner"),
+        IfInState(
+            "draft",
+            then_=[PrimaryCommunityMembers()],
+        ),
+    ]
+
+    can_read = can_read_generic + [
+        IfInState(
+            "published",
+            then_=[
+                IfRestricted(
+                    "record",
+                    then_=[PrimaryCommunityMembers()],
+                    else_=[AnyUser()],
+                )
+            ],
+        ),
+    ]
+
+    # Restricted files are accessible only to the record owner, curator, and community owner
+    can_read_files = can_read_generic + [
+        IfInState(
+            "published",
+            then_=[
+                IfRestricted(
+                    "files",
+                    then_=[
+                        RecordOwners(),
+                        CommunityRole("curator"),
+                        CommunityRole("owner"),
+                    ],
+                    else_=[AnyUser()],
+                )
+            ],
+        ),
+    ]
+
```

Comment:
If the record is in draft status, the can_read_generic rule states that primary community members can read it. Once it is published, however, only the owner, the community curator, or the community owner can read it. Is this truly what NTK wants, or do they want owners and curators to be the only readers at all times? Please clarify with NTK and record the final decision in NTK-79.